go/internal/flag/flag.go

```diff
@@ -72,13 +72,11 @@ func Parse(fs *flag.FlagSet) {
 
 // IsFlagProvided returns if the given flag has been provided by the user explicitly or not
 func IsFlagProvided(name string) bool {
-	found := false
-	flag.Visit(func(f *flag.Flag) {
-		if f.Name == name {
-			found = true
-		}
-	})
-	return found
+	fl := flag.Lookup(name)
+	if fl != nil {
+		return fl.Changed
```

Comment:
What are the semantics here if someone did provide a flag but gave it the default value? Does `fl.Changed` return `true` or `false` in that case? 

And if that's different from before, does that matter or not?

Reply:
`fl.Changed` is `true` even when you pass a flag with the default value. That semantic is very important for us because we want to know if a flag is explicitly passed or not. We don't care if it is the default value. I had the question in mind when I implemented this, therefore I have added 2 more tests in this PR which test this scenario of passing a flag explicitly equal to its default value.

Reply:
As an extra check, I had also changed the `vtctld` binary locally to print the lock timeout value and here are the results - 
```
~/vitess   flag-lookup-race ±  vtctld
E0105 12:53:08.941311   51261 main.go:55] lock-timeout value - 45s
~/vitess   flag-lookup-race ±  vtctld --remote_operation_timeout=20s
E0105 12:53:19.659368   51292 main.go:55] lock-timeout value - 20s
~/vitess   flag-lookup-race ±  vtctld --remote_operation_timeout=20s --lock-timeout=32s
E0105 12:53:27.063374   51321 main.go:55] lock-timeout value - 32s
~/vitess   flag-lookup-race ±  vtctld --remote_operation_timeout=20s --lock-timeout=45s
E0105 12:53:30.086307   51350 main.go:55] lock-timeout value - 45s
~/vitess   flag-lookup-race ±  vtctld --lock-timeout=45s
E0105 12:53:43.704695   51381 main.go:55] lock-timeout value - 45s
~/vitess   flag-lookup-race ±  vtctld --remote_operation_timeout=15s
E0105 12:53:56.462829   51412 main.go:55] lock-timeout value - 15s
```
This function works exactly as before.